src/odemis/driver/test/ueye_test.py

```diff
@@ -97,5 +98,63 @@ def setUp(self):
         VirtualTestCam.setUp(self)
 
 
+class TestUEyeCroppingTranspose(VirtualTestCam, unittest.TestCase):
+    """
+    Test setting up the UEye class with a transposed image and a set resolution to force cropping.
+    """
+    camera_type = CLASS
+    camera_kwargs = KWARGS
+
+    @classmethod
+    def setUpClass(cls):
+        if TEST_NOHW:
+            raise unittest.SkipTest('No camera HW present. Skipping tests.')
+
+        # set a fixed max_res smaller than the maximum supported camera resolution
+        cls.camera_kwargs['max_res'] = (1000, 1000)
+        cls.camera_kwargs['transp'] = [2, -1]
+
+        super(TestUEyeCroppingTranspose, cls).setUpClass()
+
+    @classmethod
+    def tearDownClass(cls):
+        if TEST_NOHW:
+            return
+        super(TestUEyeCroppingTranspose, cls).tearDownClass()
+
+    def setUp(self):
+        if TEST_NOHW:
+            self.skipTest("No simulator available")
+
+        VirtualTestCam.setUp(self)
+
+    def test_set_resolution(self):
+        """
+        Test if the max resolution and the transpose argument are passed and applied.
+        Also test if an acquired image is of the correct dimensions and the axes are right also with binning active.
+        """
+        # assign the right resolutions from the initialization of the camera class
+        cam_resolution = self.camera._sensor_res
+        trans_resolution = (self.camera_kwargs['max_res'][1], self.camera_kwargs['max_res'][0])
+        set_resolution = self.camera._metadata[model.MD_SENSOR_SIZE]
+
+        # check if the resolution is not the same, own cam res should be higher
+        self.assertNotEqual(self.camera_kwargs['max_res'], cam_resolution)
+        # check if the transpose parameter is applied and axes are switched
+        self.assertEqual(cam_resolution, tuple(reversed(set_resolution)))
+
+        # be sure to have binning set to base value
+        self.camera.binning.value = (1, 1)
+        # check if the acquired image is of the right size
+        im = self.camera.data.get()
+        self.assertEqual(im.shape, trans_resolution)
```

Comment:
The shape of a numpy array is always reversed compared to the user resolution, because the numpy array is encoded as YX (that's numpy's convention!).
So it's expected that the shape is reversed. I'd suggest that instead of `trans_resolution`, you use the expected resolution, compare like that:
```
self.assertEqual(im.shape, expected_res[::-1])
```